### PR TITLE
moves jwt to headers on from/ffrom level

### PR DESCRIPTION
**Pull request by MalkoVl**
Before the sending a URL to the native functions we split it to the URL without tokens and headers with tokens. It allows us to keep tokens safe and to prevent opening the urls (from the Network Dev Tool) on the new tab of the browser.

It is the most common implementation of moving jwt (and other token) to the request headers.

Before this changes we had to do this manually in every place in the code (for every media elements in the UI). Now, we don't have to worry about it, everything work automatically.

To get it works we moved some functions from other repos to the flow9. Is it OK?

And also was added new style (to the Picture and Video elements) to prevent parsing URL and leave it as is. SharedUrl().
It works for both From and FForm UIs.

Related PR in other repos:
[Innovation](https://github.com/area9innovation/innovation/pull/3991)
[Lyceum](https://git.area9lyceum.com/Lyceum/lyceum/pulls/8779)

### CODE REVIEW

File: lib/fform/fformutils.flow [hunk 1]
```diff
@@ -2237,3 +2244,100 @@ addFAccessProperties(properties : [FAccessProperty], zorder : Transform<[int]>)
...
+
+						FPicture(url, m.size, concat(style2, [headersStyle]));
+					}
+				);
+			}
+		);
+	}
+}
+
+// FPictureApplyJwtExtraction(m : FPicture) -> FPicture {
```
Comment (yegordovganich) [inline lib/fform/fformutils.flow:2310]: make sense to remove this code?

File: lib/formats/uri/url_utils.flow [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+setUnsetRequestHeadersLess() -> () -> void {
+	requestHeadersLessR := true;
+	\-> requestHeadersLessR := false;
+}
+
+isPhpUrl(url : string) -> bool {
+	strContains(url, ".php?");
+}
+
+isPhpUrlImproved(url : string) -> bool {
```
Comment (yegordovganich) [inline lib/formats/uri/url_utils.flow:71]: I understand that you just moved this file from innovation repo, but this fn naming looks strange. It doesn't look like it really check for php

**Review by yegordovganich — APPROVED** [review]

**Review by NikitaFil — APPROVED** [review]
Generally, looks fine.

But maybe get rid of some types and not reimplement *ApplyJwtExtraction four times?

File: lib/fform/renderfform.flow [hunk 3]
```diff
@@ -189,7 +189,9 @@ renderFForm(rform : FForm, zorder : Transform<[int]>) -> FRenderResult {
 				renderFParagraph(text, style);
 			}
 		}
-		FPicture(url_, wh, style): {
+		FPicture(url2, wh, style2): {
+			f = FPictureApplyJwtExtraction(FPicture(url2, wh, style2));
```
Comment (NikitaFil) [inline lib/fform/renderfform.flow:193]: Also, let's make it FPictureApplyJwtExtraction(rForm) and don't create additional object.

**Pull request merged**